qualtran/bloqs/rotations/phase_gradient.py

```diff
@@ -222,7 +227,12 @@ def __attrs_post_init__(self):
 
     @cached_property
     def signature(self):
-        return Signature.build(x=self.inp_bitsize, phase_grad=self.phase_bitsize)
+        return Signature(
+            [
+                Register("x", QUInt(self.inp_bitsize)),
+                Register("phase_grad", QUFxP(0, self.phase_bitsize)),
+            ]
+        )
```

Comment:
I'm confused how / why the tests pass. This definitely looks like a bug? 

`AddScaledValIntoPhaseReg` exepcts a `phase_grad` register of type `QUFxp` with `phase_bitsize` fractional bitsize and `0` word bitsize. 

This bloq is called from ` HammingWeightPhasingViaPhaseGradient.build_decompose_bloq` which expects `phase_grad` as a `QUInt(self.b_grad)` and does no explicit cast before calling 

```
        out, phase_grad = bb.add(
            AddScaledValIntoPhaseReg(
                self.bitsize.bit_length(), self.b_grad, self.exponent / 2, self.b_phase
            ),
            x=out,
            phase_grad=phase_grad,
        )
```

Do you know what's going on here?

Reply:
Well they pass because there is no type checking on the registers currently and they both have the same bitsize (num_qubits). What should the correct type be?

For migration I'm mainly focussed on removing integer bitsizes for building registers. Once that's complete we'll start enforcing types (and not just bitsizes) are consistent % generic bloqs with QAny registers.